docs/parameters.rst

```diff
@@ -4,24 +4,32 @@ Choosing Parameters
 .. note::
 
   You can probably just use :class:`argon2.PasswordHasher` with its default values and be fine.
-  Only tweak these if you've determined using :doc:`cli` that these defaults are too slow or too fast for your use case.
+  But it's good to double check using ``argon2_cffi``'s :doc:`cli` client, whether its defaults are too slow or too fast for your use case.
 
 Finding the right parameters for a password hashing algorithm is a daunting task.
-The authors of Argon2 specified a method in their `paper <https://github.com/P-H-C/phc-winner-argon2/blob/master/argon2-specs.pdf>`_ but it should be noted that they also  mention that no value for ``time_cost`` or ``memory_cost`` is actually insecure (cf. section 6.4).
+The authors of Argon2 specified a method in their `paper <https://github.com/P-H-C/phc-winner-argon2/blob/master/argon2-specs.pdf>`_, however some parts of it have been revised in the `RFC draft`_ for Argon2 that is currently being written.
 
-#. Choose whether you want Argon2i or Argon2d (``type``).
-   If you don't know what that means, choose Argon2i (:attr:`argon2.Type.I`).
-#. Figure out how many threads can be used on each call to Argon2 (``parallelism``).
+The current recommended best practice is as follow:
+
+#. Choose whether you want Argon2i, Argon2d, or Argon2id (``type``).
+   If you don't know what that means, choose Argon2id (:attr:`argon2.Type.ID`).
+#. Figure out how many threads can be used on each call to Argon2 (``parallelism``, called "lanes" in the RFC).
    They recommend twice as many as the number of cores dedicated to hashing passwords.
+   :class:`~argon2.PasswordHasher` will *not* determine this for you and use a default value that you can find in the linked API docs.
 #. Figure out how much memory each call can afford (``memory_cost``).
-#. Choose a salt length.
-   16 Bytes are fine.
-#. Choose a hash length (``hash_len``).
-   16 Bytes are fine.
+   The RFC recommends 4 GB for backend authentication and 1 GB for frontend authentication.
+#. Select the salt length.
+   16 bytes is sufficient for all applications, but can be reduced to 8 bytes in the case of space constraints.
+#. Choose a hash length (``hash_len``, called "tag length" in the documentation).
+   16 bytes is sufficient for most applications, including key derivation.
```

Comment:
I usually tell people to derive 256 bit symmetric keys, but you don't have to edit that. 16 is fine for password verification, definitely. Since the output of the derivation function has a header in it you would probably still wring it through HKDF or something.

Reply:
I’ll rephrase it to make it about password verification.